test/allocation_macros.jl

```diff
@@ -0,0 +1,64 @@
+# Helper Macros for checking that a function doesn't allocate memory
+# https://forem.julialang.org/dpinol/detecting-test-allocated-gotchas-34op
+
+macro testnoallocations(expressions...)
+    # Uncomment the following line if non-const globals should be prohibited
+    _failIfNonConstGlobalsInExpressions(__module__, expressions...)
```

Comment:
please use our code style for all function names and variables. we don't use camel case, and avoid underscores whenever possible.